internal/tools/terraformlinter/terraform_linter.go

```diff
@@ -0,0 +1,315 @@
+					trailingNewline := false
+					if len(tokens) > 0 && tokens[0].Type == hclsyntax.TokenNewline {
+						trailingNewline = true
+					}
+					position, ok := positionMap[contents]
+					if !ok {
+						position = Ignored
+					}
+					attrs = append(attrs, tokenAttr{tokenPos: position, token: token, trailingNewline: trailingNewline})
+					skipping = false
+				}
+			}
+		}
+	}
+	return generateViolations(attrs)
+}
+
+func generateViolations(idents []tokenAttr) []*ViolationInstance {
+	var instances []*ViolationInstance
+	var lastAttr tokenAttr
+	for pos, token := range idents {
+		contents := string(token.token.Bytes)
+		switch contents {
+		// for_each and count should be at the top
+		case attrForEach:
+			fallthrough
+		case attrCount:
+			if pos != 0 {
+				instances = append(instances, &ViolationInstance{ViolationType: fmt.Sprintf(violationLeadingMetaBlockAttribute, contents), Path: token.token.Range.Filename, Line: token.token.Range.Start.Line})
+			}
+		// provider is at the top but below for_each or count if they exist
+		case attrProvider:
+			if pos > 0 && lastAttr.tokenPos != Leading {
+				instances = append(instances, &ViolationInstance{ViolationType: fmt.Sprintf(violationLeadingMetaBlockAttribute, attrProvider), Path: token.token.Range.Filename, Line: token.token.Range.Start.Line})
+			}
+		// source should be the top attribute in a module
+		case attrSource:
+			if pos != 0 {
+				instances = append(instances, &ViolationInstance{ViolationType: fmt.Sprintf(violationLeadingMetaBlockAttribute, attrSource), Path: token.token.Range.Filename, Line: token.token.Range.Start.Line})
+			}
+		case attrDependsOn:
+			// depends_on somewhere above where it should be
+			if pos < len(idents)-1 && idents[len(idents)-1].tokenPos != Trailing {
+				instances = append(instances, &ViolationInstance{ViolationType: fmt.Sprintf(violationTrailingMetaBlockAttribute, attrDependsOn), Path: token.token.Range.Filename, Line: token.token.Range.Start.Line})
+			}
+			// depends_on after lifecycle
+			if pos == len(idents)-1 && lastAttr.tokenPos == Trailing {
+				instances = append(instances, &ViolationInstance{ViolationType: fmt.Sprintf(violationTrailingMetaBlockAttribute, attrDependsOn), Path: token.token.Range.Filename, Line: token.token.Range.Start.Line})
+			}
+		case attrLifecycle:
+			// lifecycle should be last
+			if pos != len(idents)-1 {
+				instances = append(instances, &ViolationInstance{ViolationType: fmt.Sprintf(violationTrailingMetaBlockAttribute, attrLifecycle), Path: token.token.Range.Filename, Line: token.token.Range.Start.Line})
+			}
+		// All provider specific entries follow the same logic. Should be below the metadata segment and above everything else
+		case attrProviderProject:
+			fallthrough
+		case attrProviderProjectID:
+			fallthrough
+		case attrProviderFolder:
+			fallthrough
+		case attrProviderFolderID:
+			fallthrough
+		case attrProviderOrganization:
+			fallthrough
+		case attrProviderOrganizationID:
+			fallthrough
```

Comment:
You can condense this a bit (not required):

```suggestion
		case attrProviderProject,
		  attrProviderProjectID,
		  attrProviderFolder,
		  attrProviderFolderID,
		  attrProviderOrganization,
		  attrProviderOrganizationID:
			fallthrough
```

Reply:
No need for the fallthrough at all, just matching on all of the possible values. I didn't realize I could do that and makes me question what the use case for fallthrough actually is since this is much cleaner.